### PR TITLE
Don't destroy CompositingFunctors that are not ready 

### PR DESCRIPTION
**Pull request by AlanGriffiths**
Fixes: #2547

### CODE REVIEW

**Review by Saviq — CHANGES_REQUESTED** [review]

File: src/server/compositor/multi_threaded_compositor.cpp [hunk 1]
```diff
@@ -385,8 +381,21 @@ void mc::MultiThreadedCompositor::create_compositing_threads()
...
         functor->wait_until_started();
+    }
+    catch (...)
+    {
+        x = std::current_exception();
+    }
+
+    if (x)
+    {
+        rethrow_exception(x);
```
[the remark this answers is not in the document]
Comment (AlanGriffiths) [reply src/server/compositor/multi_threaded_compositor.cpp:397]: Why add noise? This is exactly what ADNL is for. The function even does what it says.

[the remark this answers is not in the document]
Comment (Saviq) [reply src/server/compositor/multi_threaded_compositor.cpp:397]: OK. Was just looking for consistency with the above. Convinced.

**Review by Saviq — APPROVED** [review]
bors merge

**Comment by Saviq** [comment]
bors retry

**Pull request merged**